src/components/AddNewComment.jsx

```diff
@@ -0,0 +1,61 @@
+import { useContext, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { addCommentToArticleByID } from "../utils/api";
+import { UserContext } from "../contexts/UserContext";
+import { Link } from "react-router-dom";
+
+function AddNewComment({ comments, setComments }) {
+  const { user } = useContext(UserContext);
+  const [newComment, setNewComment] = useState({ username: user, body: "" });
+  const [error, setError] = useState(false);
+
+  const { article_id } = useParams();
+
+  const handleChange = (event) => {
+    setNewComment((newComment) => {
+      const edit = { ...newComment };
+      edit.body = event.target.value;
+
+      return edit;
+    });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    addCommentToArticleByID(article_id, newComment)
+      .then((response) => {
+        setComments([response, ...comments]);
+      })
+
+      .catch(() => {
+        setError(true);
```

Comment:
Great that you've got an error state, you should use it to provide user feedback if the comment posting is unsuccessful for any reason (for example if the internet drops while the comment is being posted)